activation/nipost.go

```diff
@@ -388,23 +418,27 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 	})
 }
 
-// Submit the challenge to all registered PoETs.
-func (nb *NIPostBuilder) submitPoetChallenges(
+// Submit the challenge to registered PoETs with given addresses.
+func (nb *NIPostBuilder) submitPoetChallengesToAddresses(
```

Comment:
I think the logic would be simpler if this function had been responsible for finding the poet services it needs to register in. I.e. call it unconditionally and let it figure out missing registrations. It could return the registered poet services for the builder logic to use later to fetch the proof so it doesn't need to get it from the DB again. Wdyt?

Reply:
Good idea, thought about that too, but afraid, that it would be difficult to review.